File: exercism/zig/leap/leap.zig
pub fn isLeapYear(year: u32) bool {
    var year_state = false;
    if (year % 400 == 0) {
        year_state = true;
    } else if (year % 100 == 0) {
        year_state = false;
    } else if (year % 4 == 0) {
        year_state = true;
    } else {
        year_state = false;
    }
    return year_state;
}

File: exercism/swift/leap/Sources/Leap/Leap.swift
class Year {
  let year: Int
  lazy var isLeapYear = year.isMultiple(of: 4)
      && (!year.isMultiple(of: 100) || year.isMultiple(of: 400))

  init(calendarYear: Int) {
    year = calendarYear
  }

}
